packages/web/src/shared/Progress.tsx

```diff
@@ -0,0 +1,108 @@
+import { useRouter } from 'next/router'
+import * as React from 'react'
+import { StyleSheet, View } from 'react-native'
+import { colors } from 'src/styles'
+import { EffectiveTypes, getEffectiveConnection } from 'src/utils/utils'
+
+// https://nextjs.org/docs/api-reference/next/router#router-api
+enum RouterEvents {
+  routeChangeStart = 'routeChangeStart',
+  routeChangeComplete = 'routeChangeComplete',
+  routeChangeError = 'routeChangeError',
+}
+
+export function usePageTurner() {
+  const router = useRouter()
+  const [isPageTurning, setPageTurning] = React.useState(false)
+  const [hasError, setError] = React.useState(false)
+  const [route, setRoute] = React.useState('')
+  React.useEffect(() => {
+    router.events.on(RouterEvents.routeChangeStart, (url) => {
+      setPageTurning(true)
+      setError(false)
+      setRoute(url)
+    })
+    router.events.on(RouterEvents.routeChangeComplete, () => {
+      setPageTurning(false)
+      setError(false)
+    })
+    router.events.on(RouterEvents.routeChangeError, (error, url) => {
+      if (error.cancelled) {
+        setRoute(url)
+      }
+      setError(true)
+    })
+  }, [])
+
+  return { isPageTurning, hasError, route }
+}
+
+export default function Progress() {
+  const { isPageTurning, hasError, route } = usePageTurner()
+
+  if (isPageTurning) {
+    const speed = getEffectiveConnection(navigator)
+    const animationSpeed = styles[speed]
```

Comment:
as am added bonus on chromium browsers speed up or slow down the animation based on estimated download speed